runtime/lua/vim/lsp/protocol.lua

```diff
@@ -629,6 +326,733 @@ function protocol.resolve_capabilities(server_capabilities)
+  ['SASS'] = 'sass',
+  ['sass'] = 'SASS',
+  ['Scala'] = 'scala',
+  ['scala'] = 'Scala',
+  ['ShaderLab'] = 'shaderlab',
+  ['shaderlab'] = 'ShaderLab',
+  ['ShellScript'] = 'shellscript',
+  ['shellscript'] = 'ShellScript',
+  ['SQL'] = 'sql',
+  ['sql'] = 'SQL',
+  ['Swift'] = 'swift',
+  ['swift'] = 'Swift',
+  ['TypeScript'] = 'typescript',
+  ['typescript'] = 'TypeScript',
+  ['TypeScriptReact'] = 'typescriptreact',
+  ['typescriptreact'] = 'TypeScriptReact',
+  ['TeX'] = 'tex',
+  ['tex'] = 'TeX',
+  ['VisualBasic'] = 'vb',
+  ['vb'] = 'VisualBasic',
+  ['XML'] = 'xml',
+  ['xml'] = 'XML',
+  ['XSL'] = 'xsl',
+  ['xsl'] = 'XSL',
+  ['YAML'] = 'yaml',
+  ['yaml'] = 'YAML',
+}
+
+---Describes how an {@link InlineCompletionItemProvider inline completion provider} was triggered.
+---
+---@since 3.18.0
+---@proposed
+---@nodoc
+---@enum lsp.InlineCompletionTriggerKind
+protocol.InlineCompletionTriggerKind = {
+  ['Invoked'] = 1,
+  [1] = 'Invoked',
+  ['Automatic'] = 2,
+  [2] = 'Automatic',
+}
+
+---A set of predefined position encoding kinds.
+---
+---@since 3.17.0
+---@nodoc
+---@enum lsp.PositionEncodingKind
+protocol.PositionEncodingKind = {
+  ['UTF8'] = 'utf-8',
+  ['utf-8'] = 'UTF8',
+  ['UTF16'] = 'utf-16',
+  ['utf-16'] = 'UTF16',
+  ['UTF32'] = 'utf-32',
+  ['utf-32'] = 'UTF32',
+}
+
+---The file event type
+---@nodoc
+---@enum lsp.FileChangeType
+protocol.FileChangeType = {
+  ['Created'] = 1,
+  [1] = 'Created',
+  ['Changed'] = 2,
+  [2] = 'Changed',
+  ['Deleted'] = 3,
+  [3] = 'Deleted',
+}
+
+---@nodoc
+---@enum lsp.WatchKind
+protocol.WatchKind = {
+  ['Create'] = 1,
+  [1] = 'Create',
+  ['Change'] = 2,
+  [2] = 'Change',
+  ['Delete'] = 4,
+  [4] = 'Delete',
+}
+
+---The diagnostic's severity.
+---@nodoc
+---@enum lsp.DiagnosticSeverity
+protocol.DiagnosticSeverity = {
+  ['Error'] = 1,
+  [1] = 'Error',
+  ['Warning'] = 2,
+  [2] = 'Warning',
+  ['Information'] = 3,
+  [3] = 'Information',
+  ['Hint'] = 4,
+  [4] = 'Hint',
+}
+
+---The diagnostic tags.
+---
+---@since 3.15.0
+---@nodoc
+---@enum lsp.DiagnosticTag
+protocol.DiagnosticTag = {
+  ['Unnecessary'] = 1,
+  [1] = 'Unnecessary',
+  ['Deprecated'] = 2,
+  [2] = 'Deprecated',
+}
+
+---How a completion was triggered
+---@nodoc
+---@enum lsp.CompletionTriggerKind
+protocol.CompletionTriggerKind = {
+  ['Invoked'] = 1,
+  [1] = 'Invoked',
+  ['TriggerCharacter'] = 2,
+  [2] = 'TriggerCharacter',
+  ['TriggerForIncompleteCompletions'] = 3,
+  [3] = 'TriggerForIncompleteCompletions',
+}
+
+---How a signature help was triggered.
+---
+---@since 3.15.0
+---@nodoc
+---@enum lsp.SignatureHelpTriggerKind
+protocol.SignatureHelpTriggerKind = {
+  ['Invoked'] = 1,
+  [1] = 'Invoked',
+  ['TriggerCharacter'] = 2,
+  [2] = 'TriggerCharacter',
+  ['ContentChange'] = 3,
+  [3] = 'ContentChange',
+}
+
+---The reason why code actions were requested.
+---
+---@since 3.17.0
+---@nodoc
+---@enum lsp.CodeActionTriggerKind
+protocol.CodeActionTriggerKind = {
+  ['Invoked'] = 1,
```

Comment:
Is there potential to reduce the amount of bi-directional mappins? If I remember right the plan was to mostly remove those.

Reply:
Perhaps we could filter this better. I was just trying to preserve the existing behavior.